core/state_update.go

```diff
@@ -34,3 +37,108 @@ type DeclaredV1Class struct {
 	ClassHash         *felt.Felt
 	CompiledClassHash *felt.Felt
 }
+
+func (d *StateDiff) Commitment() *felt.Felt {
```

Comment:
Where would this be used?

Reply:
We will use it to verify the signature from sequencer https://community.starknet.io/t/introducing-p2p-authentication-and-mismatch-resolution-in-v0-12-2/97993#signature-on-state-diff-commitment-and-block-hash-3